eng/pipelines/templates/jobs/vmr-build.yml

```diff
@@ -361,13 +361,32 @@ jobs:
         set -ex
 
         dockerVolumeArgs="-v $(sourcesPath):/vmr"
-        poisonArg=''
+        sourceOnlyArgs=''
+        extraBuildProperties=''
 
-        if [[ '${{ parameters.enablePoison }}' == 'True' ]]; then
-          poisonArg='--poison'
+        if [[ ! -z '${{ parameters.targetOS }}' ]]; then
+          extraBuildProperties="$extraBuildProperties /p:TargetOS=${{ parameters.targetOS }}"
+        fi
+
+        if [[ ! -z '${{ parameters.targetArchitecture }}' ]]; then
+          extraBuildProperties="$extraBuildProperties /p:TargetArchitecture=${{ parameters.targetArchitecture }}"
+        fi
+
+        if [[ '${{ parameters.buildSourceOnly }}' == 'True' ]]; then
+          if [[ '${{ parameters.enablePoison }}' == 'True' ]]; then
+            sourceOnlyArgs='--poison'
+          fi
+          sourceOnlyArgs="$sourceOnlyArgs --source-only /p:SmokeTestsWarnOnSdkContentDiffs=true /p:SmokeTestsExcludeOmniSharpTests=${{ parameters.excludeOmniSharpTests }}"
+        fi
+
+        # Only use Docker when a container is specified
+        if [[ -n "${{ parameters.container }}" ]]; then
+          docker run --rm $dockerVolumeArgs -w /vmr ${{ parameters.container }} ./build.sh /bl:artifacts/log/Release/Test.binlog --test $sourceOnlyArgs $extraBuildProperties $(additionalBuildArgs) 
```

Comment:
Consider encapsulating the build command to avoid duplication between container and non-container execution..
The benefit is to ensure equality between the two and improve the maintainability of the code.